core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1992,4 +1992,32 @@ package object config {
       .version("3.1.0")
       .doubleConf
       .createWithDefault(5)
+
+  private[spark] val SHUFFLE_NUM_PUSH_THREADS =
+    ConfigBuilder("spark.shuffle.push.numPushThreads")
+      .doc("Specify the number of threads in the block pusher pool. These threads assist " +
+        "in creating connections and pushing blocks to remote shuffle services when push based " +
+        "shuffle is enabled. By default, the threadpool size is equal to the number of cores.")
+      .version("3.1.0")
+      .intConf
+      .createOptional
+
+  private[spark] val SHUFFLE_MAX_BLOCK_SIZE_TO_PUSH =
+    ConfigBuilder("spark.shuffle.push.maxBlockSizeToPush")
+      .doc("The max size of an individual block to push to the remote shuffle services when push " +
+        "based shuffle is enabled. Blocks larger than this threshold are not pushed.")
+      .version("3.1.0")
+      .bytesConf(ByteUnit.KiB)
+      .createWithDefaultString("800k")
+
+  private[spark] val SHUFFLE_MAX_BLOCK_BATCH_SIZE_FOR_PUSH =
+    ConfigBuilder("spark.shuffle.push.maxBlockBatchSize")
+      .doc("The max size of a batch of shuffle blocks to be grouped into a single push request " +
+        "when push based shuffle is enabled.")
```

Comment:
@otterc and @mridulm Can we remove this phrase here and line 2008 and line 2000?
> "when push based shuffle is enabled."

Spark configuration is based on the namespace concept and all of these subconfigurations are under `spark.shuffle.push` namespace which are only enabled when `spark.shuffle.push.enabled=true`. So, we don't need to mention this repeatedly.

Also, cc @cloud-fan

Reply:
done